packages/medusa/src/services/batch-job.ts

```diff
@@ -108,6 +108,13 @@ class BatchJobService extends TransactionBaseService {
     batchJobId: string,
     config: FindConfig<BatchJob> = {}
   ): Promise<BatchJob | never> {
+    if (!isDefined(batchJobId)) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `BatchJob id should be defined`
```

Comment:
suggestion: Instead of using `isDefined` which only check that it is not of type of undefined, I suggest that we use `isString` from the medusa util. here we still allow to provide a null value but we are expecting a string. wdyt? same, depending on your answer it needs to be updated everywhere

Reply:
In case we use the name of the parameter in the message, how could we use on cases where the parameter name is `id`? I would like to somehow keep the entity explicit.


Regarding the use of `isDefined`, for now the intention is only to avoid the `property: undefined` behave as an empty object.
I don't know if in the future someone will be able to replace a service entity and use auto-increment integers as id for example.

Reply:
the stack trace will provide context for that no?
Regarding the isDefined, we are still allowed to pass null when we expect a string to be provided, so I think it make sense to at least validate != null wdyt?

Reply:
The stack trace will provide context for any message indeed. Just wanted to find something friendly and descriptive enough.

`null` won't cause the bug to retrieve random entries. in that case the regular message saying `id null was not found` will be triggered.

Reply:
I renamed all the `id` arguments to something more meaningful and added it to the error message.